### DISCUSSION TITLE
Beatmap version manager and a "diff" like system to compare them

### DISCUSSION THREAD

**Discussion by Stoppedpuma** [ppy/osu#27499] [Ideas]
I believe this is a duplicate but I can't find the original, I'm opening a new discussion since this expands upon the idea.

This addition would allow users to keep multiple versions of a beatmap and allow them to switch between them and compare them in the editor. Normally I'd manage versions with file system snapshots or copies but this isn't possible (at least efficiently) with how files are managed now on lazer.

a (very) proof of concept design:

![Group 21](https://github.com/ppy/osu/assets/58333920/183d439f-50e2-4534-9a22-07b8bfe3bcb2)
![Group 27](https://github.com/ppy/osu/assets/58333920/093f14fe-7b63-4b25-a947-fb3fab193282)

**Comment by Rekunan** [comment]
Perhaps it could also leverage the preexisting git VCS.
